raiden/api/v1/encoding.py

```diff
@@ -35,16 +35,15 @@
 )
 
 
-# type converter for the flask route
 class HexAddressConverter(BaseConverter):
-
     def to_python(self, value):
-        value = address_decoder(value)
-        return value
+        if value[:2] == '0x':
+            return value[2:].decode('hex')
+
+        raise Exception('invalid address')
 
     def to_url(self, value):
-        value = address_encoder(value)
-        return value
+        return address_encoder(value)
 
 
 class AddressField(fields.Field):
```

Comment:
Instead @hackaugusto you need to put the logic here in the `_deserialize()` method of the `AddressField`

Also please make a test in `test_api.py` where addresses without the `0x` prefix are given as input and check that we get back an appropriate error from the API.

I did some tests and if you throw from the `_deserialize()` method we get a simple `500 Internal Server Error` return and no more details. That is not sufficient as we need to alert the user that he needs to add the 0x prefix.

The solution is to throw a specific exception and then add an error handler for that exception to the flask app as detailed [here](http://flask.pocoo.org/docs/0.12/patterns/apierrors/)

This is not gonna work for us and should find a way to alert the user that he gave invalid input.